dev/src/index.ts

```diff
@@ -298,6 +298,13 @@ export class Firestore {
    */
   private _projectId: string | undefined = undefined;
 
+  /**
+   * Set of requestTags associated with the listeners that have been registered
+   * on the client.
+   * @private
+   */
+  private registeredListeners = new Set<string>();
```

Comment:
I think you only need a counter. Requests tags are also not guaranteed to be unique, since the generator just uses whatever `rand()` implementation the system exposes.